pytensor/tensor/rewriting/linalg.py

```diff
@@ -964,3 +965,58 @@ def rewrite_cholesky_diag_to_sqrt_diag(fgraph, node):
             non_eye_input = pt.shape_padaxis(non_eye_input, -2)
 
     return [eye_input * (non_eye_input**0.5)]
+
+
+# SLogDet Rewrites
+def check_sign_det(node):
+    if not (isinstance(node.op, Elemwise) and isinstance(node.op.scalar_op, Sign)):
+        return False
+
+    return True
+
+
+def check_log_abs_det(node):
+    if not (isinstance(node.op, Elemwise) and isinstance(node.op.scalar_op, Log)):
+        return False
+
+    potential_abs = node.inputs[0].owner
+    if not (
+        isinstance(potential_abs.op, Elemwise)
+        and isinstance(potential_abs.op.scalar_op, Abs)
+    ):
+        return False
+
+    return True
+
+
+def check_log_det(node):
+    if not (isinstance(node.op, Elemwise) and isinstance(node.op.scalar_op, Log)):
+        return False
+
+    return True
+
+
+@node_rewriter(tracks=[det])
+def slogdet_specialization(fgraph, node):
+    x = node.inputs[0]
+    sign_det_x, slog_det_x = SLogDet()(x)
+    replacements = {}
+    for client in list(fgraph.clients.keys()):
```

Comment:
You should go through the clients of node.outputs[0] (det) directly, not all the clients in fgraph. so `fgraph.clients[node.outputs[0]]`